python/tank/dd_utils/dd_xplatform_utils.py

```diff
@@ -0,0 +1,88 @@
+# Confidential and Proprietary Source Code
+#
+# This Digital Domain Productions, Inc. ("DDPI") source code, including without
+# limitation any human-readable computer programming code and associated
+# documentation (together "Source Code"), contains valuable confidential,
+# proprietary and trade secret information of DDPI and is protected by the laws
+# of the United States and other countries. DDPI may, from time to
+# time, authorize specific employees to use the Source Code internally at DDPI's
+# premises solely for developing, updating, and/or troubleshooting the Source
+# Code. Any other use of the Source Code, including without limitation any
+# disclosure, copying or reproduction, without the prior written authorization
+# of DDPI is strictly prohibited.
+#
+# Copyright (c) [2013] Digital Domain Productions, Inc. All rights reserved.
+#
+
+# STANDARD
+import os
+import types
+
+# DD
+from dd import xplatform
+
+# SETUP LOGGING
+from ..log import LogManager
+logger = LogManager.get_logger(__name__)
+
+
+def build_path(*paths):
```

Comment:
Why not call this xjoin?

Reply:
Because it checks whether the paths exist or not.

Reply:
Ok.... `build_path` is just a little too vague for my tastes. What about `build_search_path`?

Reply:
Nevermind, I see that there is precedence for this in `dd.runtime.api.Environment.current().buildPath`

Reply:
Alright, not renaming the method name.